detail_head.php

```diff
@@ -46,15 +46,20 @@
 
 if ($action == 'setdate_livraison' && $user->rights->expedition->creer)
 {
-    //print "x ".$_POST['liv_month'].", ".$_POST['liv_day'].", ".$_POST['liv_year'];
-    $datedelivery=dol_mktime(GETPOST('liv_hour','int'), GETPOST('liv_min','int'), 0, GETPOST('liv_month','int'), GETPOST('liv_day','int'), GETPOST('liv_year','int'));
-
-    $object->fetch($id);
-    $result=$object->set_date_livraison($user,$datedelivery);
-    if ($result < 0)
-    {
-        $mesg='<div class="error">'.$object->error.'</div>';
-    }
+	//print "x ".$_POST['liv_month'].", ".$_POST['liv_day'].", ".$_POST['liv_year'];
+	$datedelivery=dol_mktime(GETPOST('liv_hour','int'), GETPOST('liv_min','int'), 0, GETPOST('liv_month','int'), GETPOST('liv_day','int'), GETPOST('liv_year','int'));
+
+	$object->fetch($id);
+
+	if (version_compare(DOL_VERSION, '14', '>=')) {
+		$result = $object->setDateLivraison($user, $datedelivery);
+	} else {
+		$result = $object->set_date_livraison($user, $datedelivery);
+	}
```

Comment:
pour l'instant DOL_VERSION est développement
A tester car fait en direct du guthub ....
```suggestion
	if (!is_callable(array($object, 'setDateLivraison'))) {
		// For Dolibarr < 14 retrocompatibility
		$result = $object->set_date_livraison($user, $datedelivery);
	} else {
		$result = $object->setDateLivraison($user, $datedelivery);
	}
```